mantle/kola/harness.go

```diff
@@ -869,7 +870,8 @@ func runProvidedTests(testsBank map[string]*register.Test, patterns []string, mu
 
 	// Ignore the error when only denied tests with Warn:true feature failed
 	if runErr != nil && allFailedTestsAreWarnOnError(testResults.getResults()) {
-		return nil
+		return cli.ErrExitWarning77
+
```

Comment:
I know this is code I originally put together, but I think we might need to rework this a bit.

This looks great but I *think* because we don't run warn tests in the rerun then there could be a case where a warn:true test fails in the main run and then it doesn't run in the rerun but all tests in the rerun pass and are allowrerunsuccess tests.

I think in that scenario we won't warn even though a warn:true test failed.

I think we either need to not `return reRunErr` on line 867 above and then add some logic here for that OR just run the `warn:true` failures in the rerun too and keep the `return reRunErr` above. WDYT?

Reply:
Actually we rerun all tests. If failed tests with `warn:true` also have some `TAG` and `--allow-rerun-success tags=TAG` is set, then on successful rerun we return `0`, or `77` otherwise.